src/components/mc/cpu/mc_cpu.c

```diff
@@ -42,6 +42,9 @@ static ucc_status_t ucc_mc_cpu_mem_free(void *ptr)
 static ucc_status_t ucc_mc_cpu_mem_type(const void *ptr,
                                         ucc_memory_type_t *mem_type)
 {
+    if (ptr == NULL) {
```

Comment:
i'm not sure about this one. mem_type check is in the critical path. you are checking one more branch just to silence the warning. It is user responsibility to pass proper pointers. Maybe just suppress with NOLINT here as well.

Reply:
It doesn't matter because this function is not used, we can remove it.

Reply:
👍